src/ivalue_manager.rs

```diff
@@ -382,15 +442,33 @@ impl<'a> WriteHolder<IValue, IValue> for IValueKeyHolderWrite<'a> {
     }
 
     fn incr_by(&mut self, path: Vec<String>, num: &str) -> Result<Number, RedisError> {
-        self.do_num_op(path, num, |i1, i2| i1 + i2, |f1, f2| f1 + f2)
+        self.do_num_op(
+            path,
+            num,
+            |i1, i2| i1.overflowing_add(i2),
+            |f1, f2| f1 + f2,
+            |u1, u2| u1.overflowing_add(u2),
+        )
     }
 
     fn mult_by(&mut self, path: Vec<String>, num: &str) -> Result<Number, RedisError> {
-        self.do_num_op(path, num, |i1, i2| i1 * i2, |f1, f2| f1 * f2)
+        self.do_num_op(
+            path,
+            num,
+            |i1, i2| i1.overflowing_mul(i2),
+            |f1, f2| f1 * f2,
+            |u1, u2| u1.overflowing_mul(u2),
+        )
     }
 
     fn pow_by(&mut self, path: Vec<String>, num: &str) -> Result<Number, RedisError> {
-        self.do_num_op(path, num, |i1, i2| i1.pow(i2 as u32), f64::powf)
+        self.do_num_op(
+            path,
+            num,
+            |i1, i2| i1.overflowing_pow(i2 as u32),
```

Comment:
Same question as above

Reply:
Although `NUMPOWBY` is deprecated, since it is still here, same behavior as add and mul